app/views/student_task/view.html.erb

```diff
@@ -1,104 +1,353 @@
+    <input id="moveRight" value="Move right" type="button">
+    <input id="Recenter" value="Recenter" type="button">
+  </div>
+</div>
+
+<% current_folder = DisplayOption.new %>
+<% current_folder.name = "" %>
+
+<% @href_arr= Array.new %>
+<!-- @href_arr is used to store all the hyperlinks for each visualized object -->
+<% @duedates = DueDate.where("parent_id = #{@assignment.id}")
+   @visualization_data = @duedates.map do |due|
+     @href_arr.push(""); #empty hyperlink as we do not provide hyperlinks for submissions/reviews
+     if due.deadline_type_id.eql? 1
+       { :id => due.id, :start=> due.due_at, :className => "submissionDue", :content => "Round "+(due.round.to_s)+"<split>Submission due by "+'<br>'+due.due_at.strftime("%m/%d/%Y at %I:%M %p") }
+     else
+       { :id => due.id, :start=> due.due_at, :className => "reviewDue", :content => "Round "+(due.round.to_s)+"<split>Review due by "+'<br>'+due.due_at.strftime("%m/%d/%Y at %I:%M %p") }
+     end
+   end
+%>
+
+<!-- display only if submissions are made-->
+<% unless @team.nil?
+     @submissions = SubmissionRecord.find_by_sql"select * from  submission_records where assignment_id=#{@assignment.id} and team_id=#{@team.id} and content NOT IN (select content from submission_records where assignment_id=#{@assignment.id} and team_id=#{@team.id} and UPPER(operation) Like 'REMOVE%')"
+
+     @visualization_data += @submissions.map do |submission|
+       #display_directory_tree(participant, files, true).html_safe
+       if (submission.operation).eql?('Submit File')
+         file = submission.content
+         ret=""
+         if File.exist?(file) && File.directory?(file)
+           ret += link_to File.basename(file), :controller => 'submitted_content', :action => 'edit', :id => participant.id, "current_folder[name]" => file
+         else
+           ret += "\n      "
+           ret += link_to File.basename(file), :controller => 'submitted_content',
+                          :action => 'download',
+                          :id => @participant.id,
+                          :download => File.basename(file),
+                          "current_folder[name]" => File.dirname(file)
+         end
+         @href_arr.push(ret.split('"')[1])
+         #only file name instead of entire relative path need to be displayed on timeline. Hence we push the same in content, appending created time to it
+         { :id => submission.id, :start=> submission.created_at, :className=> "fileUpload", :content => (submission.content).split('/')[-1]+'<split>'+submission.created_at.strftime("%m/%d/%Y at %I:%M %p") }
+       else
+         @href_arr.push(submission.content)
+         { :id => submission.id, :start=> submission.created_at, :className=> "hyperlinkUpload" ,:content => submission.content+'<split>'+submission.created_at.strftime("%m/%d/%Y at %I:%M %p") }
+       end
+     end
+
+   end %>
+
+<!-- Reviews not yet started -->
+<% unless @review_mappings.nil?
```

Comment:
same problem here. i think there are too much conditional checking in the view that could be done in the controller. we should minimize the logic in the view since it's mixing with html.